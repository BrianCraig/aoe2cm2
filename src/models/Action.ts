enum Action {
    PICK = 'PICK',
    EXCLUSIVE_PICK = 'EXCLUSIVE_PICK',
    GLOBAL_PICK = 'GLOBAL_PICK',
    HIDDEN_PICK = 'HIDDEN_PICK',
    HIDDEN_EXCLUSIVE_PICK = 'HIDDEN_EXCLUSIVE_PICK',
    BAN = 'BAN',
    EXCLUSIVE_BAN = 'EXCLUSIVE_BAN',
    GLOBAL_BAN = 'GLOBAL_BAN',
    HIDDEN_BAN = 'HIDDEN_BAN',
    HIDDEN_EXCLUSIVE_BAN = 'HIDDEN_EXCLUSIVE_BAN',
    HIDDEN_GLOBAL_BAN = 'HIDDEN_GLOBAL_BAN',
    SNIPE = 'SNIPE',
    HIDDEN_SNIPE = 'HIDDEN_SNIPE',
    
    // REVEAL_PICKS = 'REVEAL_PICKS',
    // REVEAL_BANS = 'REVEAL_BANS',
    // REVEAL_SNIPES = 'REVEAL_SNIPES',
    REVEAL_ALL = 'REVEAL_ALL'
}

export default Action;