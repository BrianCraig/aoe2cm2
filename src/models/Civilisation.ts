import GameVersion from "./GameVersion";

enum Name {
    ARABIA = "Arabia",
    ARENA = "Arena",
    CENOTES = "Cenotes",
    GOLDEN_PIT = "Golden Pit",
    HIDEOUT = "Hideout",
    ISLANDS = "Islands",
    MEDITERRANEAN = "Mediterranean",
    SERENGETI = "Serengeti",
    VALLEY = "Valley",
    CROSS = "Cross",
    FORTRESS = "Fortress",
    GRAVEYARDS = "Graveyards",
    LAND_MADNESS = "Land Madness",

    HIDDEN_PICK = "HIDDEN_PICK",
    HIDDEN_BAN = "HIDDEN_BAN",
    HIDDEN_SNIPE = "HIDDEN_SNIPE",
    HIDDEN = "HIDDEN",
    RANDOM = "RANDOM"
}

class Civilisation {

    public static readonly ARABIA:Civilisation = new Civilisation(Name.ARABIA, GameVersion.DEFINITIVE_EDITION);
    public static readonly ARENA:Civilisation = new Civilisation(Name.ARENA, GameVersion.DEFINITIVE_EDITION);
    public static readonly CENOTES:Civilisation = new Civilisation(Name.CENOTES, GameVersion.DEFINITIVE_EDITION);
    public static readonly GOLDEN_PIT:Civilisation = new Civilisation(Name.GOLDEN_PIT, GameVersion.DEFINITIVE_EDITION);
    public static readonly HIDEOUT:Civilisation = new Civilisation(Name.HIDEOUT, GameVersion.DEFINITIVE_EDITION);
    public static readonly ISLANDS:Civilisation = new Civilisation(Name.ISLANDS, GameVersion.DEFINITIVE_EDITION);
    public static readonly MEDITERRANEAN:Civilisation = new Civilisation(Name.MEDITERRANEAN, GameVersion.DEFINITIVE_EDITION);
    public static readonly SERENGETI:Civilisation = new Civilisation(Name.SERENGETI, GameVersion.DEFINITIVE_EDITION);
    public static readonly VALLEY:Civilisation = new Civilisation(Name.VALLEY, GameVersion.DEFINITIVE_EDITION);
    public static readonly CROSS:Civilisation = new Civilisation(Name.CROSS, GameVersion.DEFINITIVE_EDITION);
    public static readonly FORTRESS:Civilisation = new Civilisation(Name.FORTRESS, GameVersion.DEFINITIVE_EDITION);
    public static readonly GRAVEYARDS:Civilisation = new Civilisation(Name.GRAVEYARDS, GameVersion.DEFINITIVE_EDITION);
    public static readonly LAND_MADNESS:Civilisation = new Civilisation(Name.LAND_MADNESS, GameVersion.DEFINITIVE_EDITION);

    public static readonly RANDOM:Civilisation = new Civilisation(Name.RANDOM, GameVersion.DEFINITIVE_EDITION);

    public static readonly HIDDEN_PICK: Civilisation = new Civilisation(Name.HIDDEN_PICK, GameVersion.TECHNICAL);
    public static readonly HIDDEN_BAN: Civilisation = new Civilisation(Name.HIDDEN_BAN, GameVersion.TECHNICAL);
    public static readonly HIDDEN_SNIPE: Civilisation = new Civilisation(Name.HIDDEN_SNIPE, GameVersion.TECHNICAL);
    public static readonly HIDDEN: Civilisation = new Civilisation(Name.HIDDEN, GameVersion.TECHNICAL);

    // DO NOT CHANGE THE ORDER OF ELEMENTS IN THIS ARRAY!!!
    // ONLY APPEND NEW CIVILISATIONS AT THE END!!!
    public static readonly ALL = [
        Civilisation.ARABIA,
        Civilisation.ARENA,
        Civilisation.CENOTES,
        Civilisation.GOLDEN_PIT,
        Civilisation.HIDEOUT,
        Civilisation.ISLANDS,
        Civilisation.MEDITERRANEAN,
        Civilisation.SERENGETI,
        Civilisation.VALLEY,
        Civilisation.CROSS,
        Civilisation.FORTRESS,
        Civilisation.GRAVEYARDS,
        Civilisation.LAND_MADNESS
    ];

    public readonly name:Name;
    public readonly gameVersion:GameVersion;
    public isRandomlyChosenCiv: boolean = false;

    private constructor(name:Name, gameVersion:GameVersion){
        this.name = name;
        this.gameVersion = gameVersion;
    }

}

export default Civilisation;
