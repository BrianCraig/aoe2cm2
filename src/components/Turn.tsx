import * as React from 'react';
import {default as ModelTurn} from '../models/Turn'
import '../pure-min.css'
import '../style2.css'
import {Trans, WithTranslation, withTranslation} from "react-i18next";

interface IProps extends WithTranslation {
    turn: ModelTurn;
    turnNumber: number;
    nextAction?: number;
}

interface IState {
    active: boolean;
}

class Turn extends React.Component<IProps, IState> {

    constructor(props: IProps) {
        super(props);
        this.state = {
            active: props.nextAction ? props.nextAction === props.turnNumber : 0 === props.turnNumber
        };
    }

    public componentWillReceiveProps(nextProps: Readonly<IProps>, nextContext: any): void {
        const active = nextProps.nextAction ? nextProps.nextAction === nextProps.turnNumber : false;
        this.setState({...this.state, active});
    }

    public render() {
        const turn: ModelTurn = this.props.turn;

        let prefix: string = '';
        if (turn.action.toString().includes('GLOBAL')) {
            prefix = 'g';
        } else if (turn.action.toString().includes('EXCLUSIVE')) {
            prefix = 'x';
        }

        let action: string = '?';
        for (const a of ['PICK', 'BAN', 'SNIPE', 'REVEAL']) {
            if (turn.action.toString().includes(a)) {
                action = a;
            }
        }

        let actionId: number = -1;
        if (action === 'PICK') {
            actionId = 0;
        }
        if (action === 'BAN' || action === 'SNIPE') {
            actionId = 1;
        }
        let turnClassName = 'turn-' + turn.player.toString().toLowerCase() + ' turn-do-' + actionId + ' turn-global';
        if (turn.action.includes('HIDDEN')) {
            turnClassName += ' turn-hidden';
        }
        let activeClass = '';
        if (this.state.active) {
            activeClass = 'active';
        }
        return (
            <div className="pure-u-1-24 turn">
                <div className={turnClassName}>
                    <span className={activeClass}><Trans><b>{prefix}</b>{action}</Trans></span>
                </div>
            </div>
        );
    }
}

export default withTranslation()(Turn);
