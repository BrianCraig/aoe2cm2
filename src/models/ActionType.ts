import Action from "./Action";

enum ActionType {
    PICK = "pick",
    BAN = "ban",
    SNIPE = "snipe",
    REVEAL = "reveal",
    NOTHING = "nothing"
}

export function actionTypeFromAction(action: Action) {
    switch (action) {
        case Action.PICK:
        case Action.EXCLUSIVE_PICK:
        case Action.GLOBAL_PICK:
        case Action.HIDDEN_PICK:
        case Action.HIDDEN_EXCLUSIVE_PICK:
            return ActionType.PICK;
        case Action.BAN:
        case Action.EXCLUSIVE_BAN:
        case Action.GLOBAL_BAN:
        case Action.HIDDEN_BAN:
        case Action.HIDDEN_EXCLUSIVE_BAN:
        case Action.HIDDEN_GLOBAL_BAN:
            return ActionType.BAN;
        case Action.SNIPE:
        case Action.HIDDEN_SNIPE:
            return ActionType.SNIPE;
        // case Action.REVEAL_PICKS:
        // case Action.REVEAL_BANS:
        // case Action.REVEAL_SNIPES:
        case Action.REVEAL_ALL:
            return ActionType.REVEAL;
    }
}

export default ActionType;