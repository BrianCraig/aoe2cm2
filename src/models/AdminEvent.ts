import Player from "./Player";
import Action from "./Action";

class AdminEvent {
    public readonly player: Player;
    public readonly action: Action;

    constructor(player: Player, action: Action) {
        this.player = player;
        this.action = action;
    }
}

export default AdminEvent;
