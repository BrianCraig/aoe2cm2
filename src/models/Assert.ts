import Player from "./Player";
import Action from "./Action";

export const Assert = {
    isString(arg: any): void {
        if (typeof arg !== "string") {
            throw new Error("Expected argument to be string, but was " + typeof arg);
        }
    },
    isPlayer(arg: any) {
        if (!Object.keys(Player).includes(arg)) {
            throw new Error("Expected argument to be a valid Player value, but was " + arg);
        }
    },
    isAction(arg: any) {
        if (!Object.keys(Action).includes(arg)) {
            throw new Error("Expected argument to be a valid Action value, but was " + arg);
        }
    }
};