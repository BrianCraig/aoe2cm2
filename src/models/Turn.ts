import Action from "./Action";
import Player from "./Player";

class Turn {
    public static readonly HOST_PICK = new Turn(Player.HOST, Action.PICK);
    public static readonly HOST_EXCLUSIVE_PICK = new Turn(Player.HOST, Action.EXCLUSIVE_PICK);
    public static readonly HOST_GLOBAL_PICK = new Turn(Player.HOST, Action.GLOBAL_PICK);
    public static readonly HOST_HIDDEN_PICK = new Turn(Player.HOST, Action.HIDDEN_PICK);
    public static readonly HOST_HIDDEN_EXCLUSIVE_PICK = new Turn(Player.HOST, Action.HIDDEN_EXCLUSIVE_PICK);
    public static readonly HOST_BAN = new Turn(Player.HOST, Action.BAN);
    public static readonly HOST_EXCLUSIVE_BAN = new Turn(Player.HOST, Action.EXCLUSIVE_BAN);
    public static readonly HOST_GLOBAL_BAN = new Turn(Player.HOST, Action.GLOBAL_BAN);
    public static readonly HOST_HIDDEN_BAN = new Turn(Player.HOST, Action.HIDDEN_BAN);
    public static readonly HOST_HIDDEN_EXCLUSIVE_BAN = new Turn(Player.HOST, Action.HIDDEN_EXCLUSIVE_BAN);
    public static readonly HOST_HIDDEN_GLOBAL_BAN = new Turn(Player.HOST, Action.HIDDEN_GLOBAL_BAN);
    public static readonly HOST_SNIPE = new Turn(Player.HOST, Action.SNIPE);
    public static readonly HOST_HIDDEN_SNIPE = new Turn(Player.HOST, Action.HIDDEN_SNIPE);

    public static readonly GUEST_PICK = new Turn(Player.GUEST, Action.PICK);
    public static readonly GUEST_EXCLUSIVE_PICK = new Turn(Player.GUEST, Action.EXCLUSIVE_PICK);
    public static readonly GUEST_GLOBAL_PICK = new Turn(Player.GUEST, Action.GLOBAL_PICK);
    public static readonly GUEST_HIDDEN_PICK = new Turn(Player.GUEST, Action.HIDDEN_PICK);
    public static readonly GUEST_HIDDEN_EXCLUSIVE_PICK = new Turn(Player.GUEST, Action.HIDDEN_EXCLUSIVE_PICK);
    public static readonly GUEST_BAN = new Turn(Player.GUEST, Action.BAN);
    public static readonly GUEST_EXCLUSIVE_BAN = new Turn(Player.GUEST, Action.EXCLUSIVE_BAN);
    public static readonly GUEST_GLOBAL_BAN = new Turn(Player.GUEST, Action.GLOBAL_BAN);
    public static readonly GUEST_HIDDEN_BAN = new Turn(Player.GUEST, Action.HIDDEN_BAN);
    public static readonly GUEST_HIDDEN_EXCLUSIVE_BAN = new Turn(Player.GUEST, Action.HIDDEN_EXCLUSIVE_BAN);
    public static readonly GUEST_HIDDEN_GLOBAL_BAN = new Turn(Player.GUEST, Action.HIDDEN_GLOBAL_BAN);
    public static readonly GUEST_SNIPE = new Turn(Player.GUEST, Action.SNIPE);
    public static readonly GUEST_HIDDEN_SNIPE = new Turn(Player.GUEST, Action.HIDDEN_SNIPE);
    
    public static readonly REVEAL_PICKS = new Turn(Player.NONE, Action.REVEAL_PICKS);
    public static readonly REVEAL_BANS = new Turn(Player.NONE, Action.REVEAL_BANS);
    public static readonly REVEAL_SNIPES = new Turn(Player.NONE, Action.REVEAL_SNIPES);
    public static readonly REVEAL_ALL = new Turn(Player.NONE, Action.REVEAL_ALL);


    public readonly player: Player;
    public readonly action: Action;

    constructor(player: Player, action: Action) {
        this.player = player;
        this.action = action;
    }
}

export default Turn;