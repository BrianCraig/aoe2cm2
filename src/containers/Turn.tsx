import {IStoreState} from '../types';
import {connect} from 'react-redux';
import Turn from "../components/draft/Turn";


export function mapStateToProps(state: IStoreState) {
    return {
        nextAction: state.nextAction
    };
}

export function mapDispatchToProps() {
    return {};
}

export default connect(mapStateToProps, mapDispatchToProps)(Turn);