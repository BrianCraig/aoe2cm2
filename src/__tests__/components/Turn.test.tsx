import * as enzyme from 'enzyme';
import * as React from 'react';
import Turn from "../../components/Turn";
import {default as ModelTurn} from "../../models/Turn";

it('gets rendered', () => {
    const turn = enzyme.shallow(<Turn turn={ModelTurn.HOST_PICK} turnNumber={0}/>);
    expect(turn).toMatchSnapshot();
});
