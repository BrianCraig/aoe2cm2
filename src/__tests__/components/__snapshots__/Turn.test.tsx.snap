// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`gets rendered 1`] = `
<div
  className="pure-u-1-24 turn"
>
  <div
    className="turn-host turn-do-0 turn-global"
  >
    <span
      className=""
    >
      <b />
      PICK
    </span>
  </div>
</div>
`;
